File: switch/fallthrough.c
#include <stdio.h>

void main() {
	int a = 1;	
	switch(a) {
		case 1: {
			printf("this is case1\n");
  			break;
		}
		case 2: {
			printf("this is case2\n");
		}
		default: {
			printf("this is default\n");
		}
	}
}

File: switch/fallthrough.go
package main

import (
	"fmt"
)

func main() {
	switch x := 1; x {
	case 1:
		fmt.Println("this is case1")
		fallthrough
	case 2:
		fmt.Println("this is case2")
		fallthrough
	default:
		fmt.Println("this is default")
	}
}
